src/backend/joanie/tests/core/api/order/test_pay_failed_installment_payment.py

```diff
@@ -0,0 +1,469 @@
+            f"/api/v1.0/orders/{order_with_failed_payment.id}/submit_installment_payment/",
+            data=payload,
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.NOT_FOUND)
+        self.assertEqual(response.json(), {"detail": "Credit card does not exist."})
+
+    def test_api_order_pay_failed_installment_without_passing_credit_card_id_in_payload(
+        self,
+    ):
+        """
+        Authenticated user should be able to pay for a failed install payment
+        on his order without passing the credit card id in the payload for the request.
+        This will call the `create_payment` method instead of the `create_one_click_payment`.
+        """
+        user = UserFactory(email="john.doe@acme.org")
+        product = ProductFactory(price=D("999.99"))
+        order = OrderFactory(
+            state=ORDER_STATE_FAILED_PAYMENT,
+            owner=user,
+            product=product,
+            payment_schedule=[
+                {
+                    "id": "1932fbc5-d971-48aa-8fee-6d637c3154a5",
+                    "amount": "200.00",
+                    "due_date": "2024-01-17",
+                    "state": PAYMENT_STATE_PAID,
+                },
+                {
+                    "id": "d9356dd7-19a6-4695-b18e-ad93af41424a",
+                    "amount": "300.00",
+                    "due_date": "2024-02-17",
+                    "state": PAYMENT_STATE_REFUSED,
+                },
+                {
+                    "id": "9fcff723-7be4-4b77-87c6-2865e000f879",
+                    "amount": "300.00",
+                    "due_date": "2024-03-17",
+                    "state": PAYMENT_STATE_PENDING,
+                },
+                {
+                    "id": "168d7e8c-a1a9-4d70-9667-853bf79e502c",
+                    "amount": "199.99",
+                    "due_date": "2024-04-17",
+                    "state": PAYMENT_STATE_PENDING,
+                },
+            ],
+        )
+        InvoiceFactory(order=order)
+        token = self.generate_token_from_user(user)
+
+        response = self.client.post(
+            f"/api/v1.0/orders/{order.id}/submit_installment_payment/",
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        self.assertEqual(response.json()["provider_name"], "dummy")
+        self.assertEqual(
+            response.json()["url"],
+            "https://example.com/api/v1.0/payments/notifications",
+        )
+
+    def test_api_order_pay_failed_installment(self):
+        """
+        Authenticated user should be able to pay for a failed install payment
+        on his order when the installment is in state 'PAYMENT_STATE_REFUSED',
+        meaning that the order should be in state 'ORDER_STATE_FAILED_PAYMENT'.
```

Comment:
You should mention that the main topic of this test is to check that with a provided credit card a one click payment should be created.
```suggestion
        Authenticated user should be able to pay for a failed installment
        on its order when the installment is in state 'PAYMENT_STATE_REFUSED',
        meaning that the order should be in state 'ORDER_STATE_FAILED_PAYMENT'.
```